### PR TITLE
feat: scatterplot

### PR DESCRIPTION
**Pull request by ejfox**

### CODE REVIEW

**Comment by curran** [comment]
LGTM! Would be great to get this merged so I can branch from here and take on the TOPOJSON issue.

**Comment by curran** [comment]
This is what page load looks like for me.

image

Would suggest to filter out the NaNs so we don't get that clump at (0, 0) upper left corner.

**Comment by curran** [comment]
Also, the label placement could use some love, at least so they are not overlapping with the tick mark labels.

**Comment by curran** [comment]
On the buttons, `user-select: none;` would be good to add in the CSS.

image

File: components/DataTable.vue [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        </tr>
+      </tfoot>
+
+    </table>
+</template>
+<script setup>
+import * as d3 from 'd3'
+defineProps(['sortedDataByGrainType', 'totalHarvestedArea', 'totalYield', 'totalVolume'])
+
+const numberFormat = (number) => {
```
Comment (curran) [inline components/DataTable.vue:41]:
```suggestion
const numberFormat = d3.format(",.0f");
```

File: pages/map.vue [hunk 2]
```diff
@@ -117,11 +215,18 @@ function initMap(geographicData) {
...
-
+      const shapeName1 = normalizeOblastName(d.properties.name_1);
+      // console.log(shapeName1, parsedDataByName)
       const oblastData = parsedDataByName.value[shapeName1];
       const shapeValue = oblastData ? oblastData[valueKey.value] : 0;
 
+      // console.log('🌻', shapeName1, oblastData, shapeValue)
+      // console.log('💯', shapeValue)
+      // console.log(oblastData[valueKey.value])
+      if (!oblastData) {
```
Comment (curran) [inline pages/map.vue:226]:
```suggestion
```

File: pages/map.vue [hunk 3]
```diff
@@ -145,11 +250,32 @@ function updateMap(geographicData) {
       const shapeValue = oblastData ? oblastData[valueKey.value] : 0;
 
       if (shapeValue) return valueColorScale.value(+shapeValue);
-      else return '#FFF'
+      else return '#CCC'
     })
 
 }
 
+// A germ of an idea: -- Curran
```
Comment (curran) [inline pages/map.vue:258]: We can remove these dated comments.

File: pages/scatterplot.vue [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+function updateScatterplot() {
+  const svg = d3.select(scatterplotSvg.value)
+
+  // Get the extent of the scatterplot X and Y in the data
+  const xExtent = d3.extent(dataByGrainType.value.get(activeGrainType.value), (d) => d[scatterplotX.value])
+
+  const yExtent = d3.extent(dataByGrainType.value.get(activeGrainType.value), (d) => d[scatterplotY.value])
+
+  // create the Y and X axes for the scatterplot
+  const x = d3.scaleLinear()
```
Comment (curran) [inline pages/scatterplot.vue:199]: Something doesn't sit right about this pattern, where the logic for setting up the scales is duplicated, once here and once in the init function. If this ever needs to change, it needs to be changed in two places.

File: pages/scatterplot.vue [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+    .attr("transform", d => `translate(${x(d[scatterplotX.value])},${y(d[scatterplotY.value])})`)
+    .transition()
+    .duration(1000)
+
+  // update axes and axes labels
+  const xAxis = d3.axisBottom(x)
+  const yAxis = d3.axisLeft(y)
+
+  // clear old axes
+  svg.select(".x-axis").remove()
```
Comment (curran) [inline pages/scatterplot.vue:231]: We could get rid of these `.remove()` lines using the following:

```js
  const xAxisGroup = svg.selectAll("g.x-axis").data([null]).join("g")
    .classed("x-axis", true)
    .attr("transform", `translate(0,${height - margin.bottom})`)
    .call(xAxis)
```

File: pages/scatterplot.vue [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+  svg.select(".y-axis").remove()
+
+  // add new axes
+  const xAxisGroup = svg.append("g")
```
Comment (curran) [inline pages/scatterplot.vue:235]:
```suggestion
  const xAxisGroup = svg.selectAll("g.x-axis").data([null]).join("g")
```

File: pages/scatterplot.vue [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+    .classed("x-axis", true)
+    .attr("transform", `translate(0,${height - margin.bottom})`)
+    .call(xAxis)
+
+  const yAxisGroup = svg.append("g")
```
Comment (curran) [inline pages/scatterplot.vue:240]:
```suggestion
  const yAxisGroup = svg.selectAll("g.y-axis").data([null]).join("g")
```

**Review by curran — APPROVED** [review]
LGTM! Added some ideas for simplifications.

File: pages/scatterplot.vue (hunk 4 above)
Comment (ejfox) [reply pages/scatterplot.vue:199]: I concur, gonna refactor

File: pages/scatterplot.vue (hunk 5 above)
Comment (ejfox) [reply pages/scatterplot.vue:231]: I find this kind of confusing, whereas .remove is very clear

**Pull request merged**